### PR TITLE
Defer dot import

### LINKED ISSUE #1899: Importing iris shouldn't result in a subprocess.check_call
https://github.com/SciTools/iris/pull/1266 introduced an automatic check_call to identify the dot executable. Because of our import machinery, this ends up getting done when ```import iris``` is run. Clearly that is not desirable - ideally it should only run when we actually *want* dot to work.

### PR DESCRIPTION
**Pull request by rhattersley**
Fixes #1899

### CODE REVIEW

File: lib/iris/io/__init__.py [hunk 1]
```diff
@@ -227,6 +227,22 @@ def load_http(urls, callback):
             yield cube
 
 
+def _dot_save(cube, target):
```
Comment (pelson) [inline lib/iris/io/__init__.py]: ```**kwargs``` missing.

**Comment by pelson** [comment]
:+1: in principle.

File: lib/iris/io/__init__.py (hunk 1 above)
Comment (rhattersley) [inline lib/iris/io/__init__.py]: `iris.fileformats.dot.save` doesn't take any keyword arguments, so perhaps better to just get rid of them from the call a couple of lines further down from here.

Comment (pelson) [inline lib/iris/io/__init__.py]: Yep, that sounds sensible.

Comment (rhattersley) [inline lib/iris/io/__init__.py]: Done.

**Pull request merged**